### DISCUSSION TITLE
MediaPicker  CapturePhotoAsync  need parameter for rear or front camera

### DISCUSSION THREAD

**Discussion by bhuwancb99** [dotnet/maui#25033] [Ideas]
### Description

In the MediaPicker, the CapturePhotoAsync method currently does not provide an option to select either the rear or front camera. Please consider adding a parameter inside MediaPickerOptions to specify the desired camera type (front or back). This would make it easier to use the camera.

In my project, I need to use the front camera for taking pictures when the user punches in the system. I've tried multiple approaches without success. I also attempted the following code for Android:

`intent.PutExtra("android.intent.extras.CAMERA_FACING", 1);`

However, it did not work. I want this same functionality for both Android and iOS, as I plan to integrate it into my two mobile applications.

### Public API Changes

`new MediaPickerOptions
                    {
                        Title = GlobalResource.SELECT_A_IMAGE,
                        CameraLocation=CameraLocation.Front
                    })`

### Intended Use-Case

`FileResult photo = await MediaPicker.Default.CapturePhotoAsync(new MediaPickerOptions
{
    Title = GlobalResource.SELECT_A_IMAGE
});`

**Comment by bhuwancb99** [comment]
Hello, any update on this?

**Reply by ghost** [reply]
You can read more about this [here](https://github.com/dotnet/maui/issues/29075). For now, I like this project [BarcodeScanning.Native.Maui](https://github.com/afriscic/BarcodeScanning.Native.Maui?tab=readme-ov-file), in source code you can find front/back camera [option](https://github.com/afriscic/BarcodeScanning.Native.Maui/blob/master/BarcodeScanning.Native.Maui/CameraView.cs). I found one [discussion which might help you too](https://github.com/dotnet/maui/discussions/9520). [We tested also this library](https://github.com/hjam40/Camera.MAUI), but the first library I posted works better for QR code reading, at least for us.
